### PR TITLE
MODULES-2179: Implement SetEnvIfNoCase

### PR DESCRIPTION
**Pull request by jlambert121**

### CODE REVIEW

**Comment by jlambert121** [comment]
It appears the broken task is unrelated to these changes.  Let me know if you want me to rebase.

File: spec/defines/vhost_spec.rb [hunk 1]
```diff
@@ -465,7 +466,12 @@
       it { is_expected.to contain_concat__fragment('rspec.example.com-rewrite') }
       it { is_expected.to contain_concat__fragment('rspec.example.com-scriptalias') }
       it { is_expected.to contain_concat__fragment('rspec.example.com-serveralias') }
-      it { is_expected.to contain_concat__fragment('rspec.example.com-setenv') }
+      it { is_expected.to contain_concat__fragment('rspec.example.com-setenv').with_content(
+              /SetEnv FOO=\/bin\/true/) }
+      it { is_expected.to contain_concat__fragment('rspec.example.com-setenv').with_content(
+              /SetEnvIf Request_URI "\\\.gif\$" object_is_image=gif/) }
+      it { is_expected.to contain_concat__fragment('rspec.example.com-setenv').with_content(
+              /SetEnvIfNoCase REMOTE_ADDR \^127.0.0.1 localhost=true/) }
```
Comment (igalic) [inline spec/defines/vhost_spec.rb]: please use `%r{}` we don't need the tooth`\/\/\/`picks

**Comment by jlambert121** [comment]
@igalic updated and rebased with test fixes.

**Comment by tphoney** [comment]
@igalic are you happy with the change ?

**Comment by igalic** [comment]
sooooooo happy

**Comment by TJM** [comment]
Thanks Justin!

**Pull request merged**